MOCC/src/configurations/Comm-Layer/src/server.py

```diff
@@ -0,0 +1,428 @@
+        """
+        raise NotImplementedError()
+
+    def get_non_auto_polled_cmd(self):
+        """
+        Returns a COPY of the list of non automatic polled commands
+        :return seq<str>: a COPY of the list of non automatic polled commands
+        """
+        raise NotImplementedError()
+
+    def get_poll_old_factor(self):
+        """
+        Returns the poll old factor
+        :return int: the poll old factor
+        """
+        raise NotImplementedError()
+
+    def get_poll_ring_depth(self):
+        """
+        Returns the poll ring depth
+        :return int: the poll ring depth
+        """
+        raise NotImplementedError()
+
+    def get_polled_attr(self):
+        """
+        Returns a COPY of the list of polled attributes
+        :return seq<str>: a COPY of the list of polled attributes
+        """
+        raise NotImplementedError()
+
+    def get_polled_cmd(self):
+        """
+        Returns a COPY of the list of polled commands
+        :return seq<str>: a COPY of the list of polled commands
+        """
+        raise NotImplementedError()
+
+    def get_prev_state(self):
+        """
+        Get a COPY of the device’s previous state.
+        :return DevState: the device's previous state
+        """
+        raise NotImplementedError()
+
+    def get_state(self):
+        """
+        Get a COPY of the device's state
+        :return DevState: Current device state
+        """
+        raise NotImplementedError()
+
+    def get_status(self):
+        """
+        Get a COPY of the device status.
+        :return str: the device status
+        """
+        return self.status
+
+    def info_stream(self, msg):
+        """
+        Sends the given message to the info stream.
+        :param msg str: the given message to the info
+        """
+        raise NotImplementedError()
+
+    def init_device(self):
+        """"""
+        raise NotImplementedError()
+
+    def initialize_dynamic_attributes(self):
+        """
+        Method executed at initializion phase to create dynamic attributes. 
+        Default implementation does nothing. Overwrite when necessary.
+        """
+        raise NotImplementedError()
+
+    def is_device_locked(self):
+        """
+        Returns if this device is locked by a client.
+        :return bool: True if it is locked or False otherwise
+        """
+        raise NotImplementedError()
+
+    def is_polled(self):
+        """
+        Returns if it is polled
+        :return bool: True if it is polled or False otherwise
+        """
+    
+    def is_there_subscriber(self, att_name, event_type):
+        """
+        Check if there is subscriber(s) listening for the event.
+        :param str att_name: the attribute name
+        :param EventType event_type: the event type
+        :return bool: True if there is at least one listener or False otherwise
+        """
+        raise NotImplementedError()
+
+    def push_archive_event(self, **kwargs):
```

Comment:
[comment 1]: why are there parameters in the docstring that are not in the signature?

Reply:
These methods have multiple prototypes http://pytango.readthedocs.io/en/stable/server_api/server.html#tango.server.Device.push_archive_event
So I thought this is the best way of taking in multiple arguments

Reply:
I think it'd be better to make the parameters explicit.
Just put a default value for the optional ones, like `attr_name=None`.
You can use `None` for all optional parameters for now.
In the case of `push_archive_event`, all versions of the method take the `attr_name` parameter, so it is not optional, must be provided by the user.
All of the above is also valid for the other two methods, `push_change_event` and `push_event`.

Reply:
OK. Will change.